mllib/src/main/scala/org/apache/spark/ml/Transformer.scala

```diff
@@ -90,7 +90,7 @@ abstract class UnaryTransformer[IN, OUT, T <: UnaryTransformer[IN, OUT, T]]
    * account of the embedded param map. So the param values should be determined solely by the input
    * param map.
    */
-  protected def createTransformFunc: IN => OUT
+  protected val createTransformFunc: (T, IN) => OUT
```

Comment:
The UDF obtained with `def createTransformFunc` can't work with runtime mirror. I tried many times. Only make it as a `val` can work.

As `Transformer` can update its params, we need to pass in `T` to UDF to get the updated params.